Description: Add --enable-examples-dynamic-linking for dynamically linking
 examples against libndpi.
Author: Ludovico Cavedon <cavedon@debian.org>

Index: ndpi/configure.ac
===================================================================
--- ndpi.orig/configure.ac
+++ ndpi/configure.ac
@@ -79,6 +79,11 @@ LIBS=$OLD_LIBS
 
 AC_CHECK_LIB(pthread, pthread_setaffinity_np, AC_DEFINE_UNQUOTED(HAVE_PTHREAD_SETAFFINITY_NP, 1, [libc has pthread_setaffinity_np]))
 
+AC_ARG_ENABLE([examples-dynamic-linking],
+             [AS_HELP_STRING([--enable-examples-dynamic-linking],
+                             [dynamically link examples against libndpi (default is no)])])
+AM_CONDITIONAL(EXAMPLES_DYNAMIC_LINK, [test "x$enable_examples_dynamic_linking" = "xyes"])
+
 AC_CONFIG_FILES([Makefile src/lib/Makefile example/Makefile libndpi.pc])
 AC_CONFIG_HEADERS(config.h)
 AC_SUBST(JSON_C_CFLAGS)
Index: ndpi/example/Makefile.am
===================================================================
--- ndpi.orig/example/Makefile.am
+++ ndpi/example/Makefile.am
@@ -4,7 +4,9 @@ AM_CPPFLAGS = -I$(top_srcdir)/src/includ
 AM_CFLAGS = @PTHREAD_CFLAGS@
 
 LDADD = $(top_builddir)/src/lib/libndpi.la @JSON_C_LIBS@ @PTHREAD_LIBS@ @PCAP_LIB@
+if !EXAMPLES_DYNAMIC_LINK
 AM_LDFLAGS = -static
+endif
 
 ndpiReader_SOURCES = ndpiReader.c ndpi_util.c
 
